src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/GenerateAzureVpnClientPackage.cs

```diff
@@ -65,10 +78,164 @@ public override void ExecuteCmdlet()
+            url = url.Replace("{subscriptionId}", Uri.EscapeDataString(this.NetworkClient.NetworkManagementClient.SubscriptionId));
+            url += "?" + string.Join("&", string.Format("api-version={0}", Uri.EscapeDataString(apiVersion)));
+
+            // Create HTTP transport objects
+            HttpRequestMessage httpRequest = new HttpRequestMessage();
+            httpRequest.Method = new HttpMethod("POST");
+            httpRequest.RequestUri = new Uri(url);
+            // Set Headers
+            httpRequest.Headers.TryAddWithoutValidation("x-ms-client-request-id", Guid.NewGuid().ToString());
+
+            // Serialize Request
+            string requestContent = JsonConvert.SerializeObject(parameters, this.NetworkClient.NetworkManagementClient.SerializationSettings);
+            httpRequest.Content = new StringContent(requestContent, Encoding.UTF8);
+            httpRequest.Content.Headers.ContentType = MediaTypeHeaderValue.Parse("application/json; charset=utf-8");
+
+            // Set Credentials
+            if (this.NetworkClient.NetworkManagementClient.Credentials != null)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                await this.NetworkClient.NetworkManagementClient.Credentials.ProcessHttpRequestAsync(httpRequest, cancellationToken).ConfigureAwait(false);
+            }
+            // Send Request
+            cancellationToken.ThrowIfCancellationRequested();
+            HttpClient httpClient = new HttpClient();
+            HttpResponseMessage httpResponse = await httpClient.SendAsync(httpRequest, cancellationToken).ConfigureAwait(false);
+
+            HttpStatusCode statusCode = httpResponse.StatusCode;
+            cancellationToken.ThrowIfCancellationRequested();
+            if ((int)statusCode != 202)
+            {
+                string responseContent = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
+                throw new Exception(string.Format("Get-AzureRmVpnClientPackage Operation returned an invalid status code '{0}' with Exception:{1}",
+                    statusCode, string.IsNullOrEmpty(responseContent) ? "NotAvailable" : responseContent));
+            }
+
+            // Create Result
+            var result = new AzureOperationResponse<string>();
+            result.Request = httpRequest;
+            result.Response = httpResponse;
+            if (httpResponse.Headers.Contains("x-ms-request-id"))
+            {
+                result.RequestId = httpResponse.Headers.GetValues("x-ms-request-id").FirstOrDefault();
+            }
+            else
+            {
+                throw new Exception(string.Format("Get-AzureRmVpnClientPackage Operation Failed as no valid status code received in response!"));
+            }
+
+            string operationId = result.RequestId;
+            #endregion
+
+            #region 2. Wait for Async operation to succeed 
+
+            Thread.Sleep(TimeSpan.FromSeconds(60));
```

Comment:
So it seems like here, you should use a while loop waiting for the response to come back as a 200.  Also, please use Microsoft.WindowsAzure.Commands.Utilities.CommonTestMockSupport.Delay  instead of calling Thread.Sleep directly, as this  will avoid delays when running mocked.